src/main/java/org/jenkinsci/plugins/saml/BundleKeyStore.java

```diff
@@ -0,0 +1,277 @@
+package org.jenkinsci.plugins.saml;
+
+import sun.security.x509.AlgorithmId;
+import sun.security.x509.CertificateAlgorithmId;
+import sun.security.x509.CertificateSerialNumber;
+import sun.security.x509.CertificateValidity;
+import sun.security.x509.CertificateVersion;
+import sun.security.x509.CertificateX509Key;
+import sun.security.x509.X500Name;
+import sun.security.x509.X509CertImpl;
+import sun.security.x509.X509CertInfo;
```

Comment:
> I will have to implement it

Yes. But I believe it is the right thing to do.

> the only way to create a certificate object is from an ASN1 input stream so I will have also to implement a weird code with temp files

[It does not seem like that](https://stackoverflow.com/questions/29852290/self-signed-x509-certificate-with-bouncy-castle-in-java).

In any case, using `sun.*` is not an option IMO ([more here](http://www.oracle.com/technetwork/java/faq-sun-packages-142232.html)).